### PR TITLE
Security and compliance

### PR DESCRIPTION
**Pull request by NikCharlebois**
#### Pull Request (PR) description
Introduces Support for S&C

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

**Comment by NikCharlebois** [comment]
@ykuijs Code review per favor :)?

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[ClassVersion("1.0.0.0"), FriendlyName("SCRetentionCompliancePolicy")]
+class MSFT_SCRetentionCompliancePolicy : OMI_BaseResource
+{
+    [Key, Description("The Name parameter specifies the unique name of the retention policy.")] String Name;
+    [Write, Description("Specify if this policy should exist or not."), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
+    [Write, Description("The Comment parameter specifies an optional comment.")] String Comment;
+    [Write, Description("")] String DynamicScopeLocation[];
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    [Write, Description("")] Boolean Enabled;
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    [Write, Description("The PublicFolderLocation parameter specifies that you want to include all public folders in the retention policy. You use the value All for this parameter.")] String PublicFolderLocation[];
+    [Write, Description("The RestrictiveRetention parameter specifies whether Preservation Lock is enabled for the policy.")] Boolean RestrictiveRetention;
+    [Write, Description("The SharePointLocation parameter specifies the SharePoint Online sites to include. You identify the site by its URL value, or you can use the value All to include all sites.")] String SharePointLocation[];
+    [Write, Description("This parameter specifies the SharePoint Online sites to exclude when you use the value All for the SharePointLocation parameter. You identify the site by its URL value.")] String SharePointLocationException[];
+    [Write, Description("The SkypeLocation parameter specifies the Skype for Business Online users to include in the policy.")] String SkypeLocation[];
+    [Write, Description("This parameter is reserved for internal Microsoft use.")] String SkypeLocationException[];
+    [Write, Description("")] String TeamsChannelLocation[];
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    [Write, Description("")] String TeamsChannelLocationException[];
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    [Write, Description("")] String TeamsChatLocation[];
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    [Write, Description("")] String TeamsChatLocationException[];
```
Comment (ykuijs) [inline Modules/Office365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.schema.mof]: Please add a description for this parameter

File: Modules/Office365DSC/Modules/Office365DSCUtil.psm1 [hunk 7]
```diff
@@ -1508,6 +1513,28 @@ function Test-TeamsServiceConnection
...
+        $GlobalAdminAccount
+    )
+    $VerbosePreference = 'SilentlyContinue'
+    $WarningPreference = "SilentlyContinue"
+
+    $Global:SessionSecurityCompliance = Get-PSSession | Where-Object{$_.ComputerName -like "*.ps.compliance.protection.outlook.com"}
+    if ($null -eq $Global:SessionSecurityCompliance)
+    {
+        Write-Verbose "Session to Security & Compliance already exists, re-using existing session"
+        $Global:SessionSecurityCompliance = New-PSSession -ConfigurationName "Microsoft.Exchange" -ConnectionUri https://ps.compliance.protection.outlook.com/powershell-liveid/ -Credential $GlobalAdminAccount -Authentication Basic -AllowRedirection
```
Comment (ykuijs) [inline Modules/Office365DSC/Modules/Office365DSCUtil.psm1]: Please move the parameters to new lines to improve readability

**Pull request merged**